### ISSUE TITLE
Blank/empty wp-login.php + checksums / force core download

### ISSUE THREAD

**Issue by mikeyb31** [wp-cli/wp-cli#4352]
Hola,

This is my first time using WP CLI and have unfortunately bumped into some strange behaviour...

```
vagrant@vagrant:/vagrant$ cat /etc/*-release
PRETTY_NAME="Debian GNU/Linux 8 (jessie)"
NAME="Debian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=debian
HOME_URL="http://www.debian.org/"
SUPPORT_URL="http://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
vagrant@vagrant:/vagrant$ php -v
PHP 7.0.19-1~dotdeb+8.1 (cli) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.0.19-1~dotdeb+8.1, Copyright (c) 1999-2017, by Zend Technologies
```

After running the following commands...

`php wp-cli.phar --path=public_html core update`
`php wp-cli.phar --path=public_html --debug plugin update --all`

...I noticed I was getting a blank white screen while trying to login to `/wp-admin`. 
I assumed this was a theme/plugin conflict but on closer inspection, I noticed in version control that the `wp-login.php` file had been completely emptied?

I assumed this was just a "glitch" so I did some research and noticed I could verify the checksums and download the original cores files but again I'm not having much luck (please see below).

```
vagrant@vagrant:/vagrant$ php wp-cli.phar --path=public_html core download --force --locale=en_GB
Downloading WordPress 4.8.1 (en_GB)...
Using cached file '/home/vagrant/.wp-cli/cache/core/wordpress-4.8.1-en_GB.tar.gz'...
Success: WordPress downloaded.
vagrant@vagrant:/vagrant$ php wp-cli.phar --path=public_html checksum core
Warning: File doesn't verify against checksum: wp-admin/includes/dashboard.php
Warning: File doesn't verify against checksum: wp-admin/includes/class-wp-posts-list-table.php
Warning: File doesn't verify against checksum: wp-admin/includes/template.php
Warning: File doesn't verify against checksum: wp-admin/includes/media.php
Warning: File doesn't verify against checksum: wp-admin/includes/meta-boxes.php
Warning: File doesn't verify against checksum: wp-admin/nav-menus.php
Warning: File doesn't verify against checksum: wp-admin/custom-header.php
Warning: File doesn't verify against checksum: wp-login.php
Warning: File doesn't verify against checksum: wp-includes/class-wp-customize-manager.php
Warning: File doesn't verify against checksum: wp-includes/formatting.php
Warning: File doesn't verify against checksum: wp-includes/class-wp-xmlrpc-server.php
Warning: File doesn't verify against checksum: wp-includes/class-phpmailer.php
Warning: File doesn't verify against checksum: wp-includes/functions.php
Warning: File doesn't verify against checksum: wp-includes/media-template.php
Warning: File doesn't verify against checksum: wp-includes/media.php
Error: WordPress install doesn't verify against checksums.
```

Any ideas/suggestions are much appreciated!

Regards
Mike

**Comment by wojsmol** [comment]
@mikeyb31 Please delete /home/vagrant/.wp-cli/cache/core/wordpress-4.8.1-en_GB.tar.gz cache file and run again second set of commands.

**Comment by gritdigital-mibu** [comment]
@wojsmol Still no luck :(

```
vagrant@vagrant:/vagrant$ rm /home/vagrant/.wp-cli/cache/core/wordpress-4.8.1-en_GB.tar.gz
vagrant@vagrant:/vagrant$ php wp-cli.phar --path=public_html core download --force --locale=en_GB --debug
Debug (bootstrap): No readable global config found (0.089s)
Debug (bootstrap): No project config found (0.09s)
Debug (bootstrap): argv: wp-cli.phar --path=public_html core download --force --locale=en_GB --debug (0.091s)
Debug (bootstrap): ABSPATH defined: /vagrant/public_html/ (0.091s)
Debug (bootstrap): Running command: core download (0.092s)
Downloading WordPress 4.8.1 (en_GB)...
md5 hash verified: 4c18244563187acdd705ea4d641e07f3
Success: WordPress downloaded.
vagrant@vagrant:/vagrant$ php wp-cli.phar --path=public_html checksum core
Warning: File doesn't verify against checksum: wp-admin/includes/dashboard.php
Warning: File doesn't verify against checksum: wp-admin/includes/class-wp-posts-list-table.php
Warning: File doesn't verify against checksum: wp-admin/includes/template.php
Warning: File doesn't verify against checksum: wp-admin/includes/media.php
Warning: File doesn't verify against checksum: wp-admin/includes/meta-boxes.php
Warning: File doesn't verify against checksum: wp-admin/nav-menus.php
Warning: File doesn't verify against checksum: wp-admin/custom-header.php
Warning: File doesn't verify against checksum: wp-login.php
Warning: File doesn't verify against checksum: wp-includes/class-wp-customize-manager.php
Warning: File doesn't verify against checksum: wp-includes/formatting.php
Warning: File doesn't verify against checksum: wp-includes/class-wp-xmlrpc-server.php
Warning: File doesn't verify against checksum: wp-includes/class-phpmailer.php
Warning: File doesn't verify against checksum: wp-includes/functions.php
Warning: File doesn't verify against checksum: wp-includes/media-template.php
Warning: File doesn't verify against checksum: wp-includes/media.php
Error: WordPress install doesn't verify against checksums.
```

**Comment by mikeyb31** [comment]
Some extra information:

## WP CLI v

```
vagrant@vagrant:/vagrant$ php wp-cli.phar --path=public_html --debug cli info
Debug (bootstrap): No readable global config found (0.087s)
Debug (bootstrap): No project config found (0.088s)
Debug (bootstrap): argv: wp-cli.phar --path=public_html --debug cli info (0.088s)
Debug (bootstrap): Running command: cli info (0.088s)
PHP binary:     /usr/bin/php7.0
PHP version:    7.0.19-1~dotdeb+8.1
php.ini used:   /etc/php/7.0/cli/php.ini
WP-CLI root dir:        phar://wp-cli.phar
WP-CLI vendor dir:      phar://wp-cli.phar/vendor
WP_CLI phar path:       /vagrant
WP-CLI packages dir:
WP-CLI global config:
WP-CLI project config:
WP-CLI version: 1.3.0
```

## WordPress v

```
vagrant@vagrant:/vagrant$ php wp-cli.phar --path=public_html --debug core version
Debug (bootstrap): No readable global config found (0.087s)
Debug (bootstrap): No project config found (0.087s)
Debug (bootstrap): argv: wp-cli.phar --path=public_html --debug core version (0.087s)
Debug (bootstrap): ABSPATH defined: /vagrant/public_html/ (0.088s)
Debug (bootstrap): Running command: core version (0.089s)
4.7.3
```

## Available updates

```
vagrant@vagrant:/vagrant$ php wp-cli.phar --path=public_html --debug core check-update
Debug (bootstrap): No readable global config found (0.084s)
Debug (bootstrap): No project config found (0.084s)
Debug (bootstrap): argv: wp-cli.phar --path=public_html --debug core check-update (0.084s)
Debug (bootstrap): ABSPATH defined: /vagrant/public_html/ (0.085s)
Debug (bootstrap): Begin WordPress load (0.086s)
Debug (bootstrap): wp-config.php path: /vagrant/public_html/wp-config.php (0.086s)
Debug (commandfactory): No doc comment for timber. (0.337s)
Debug (bootstrap): Loaded WordPress (0.417s)
Debug (bootstrap): Running command: core check-update (0.418s)
+---------+-------------+-------------------------------------------------------------------+
| version | update_type | package_url                                                       |
+---------+-------------+-------------------------------------------------------------------+
| 4.7.5   | minor       | https://downloads.wordpress.org/release/en_GB/wordpress-4.7.5.zip |
| 4.8.1   | major       | https://downloads.wordpress.org/release/en_GB/wordpress-4.8.1.zip |
+---------+-------------+-------------------------------------------------------------------+
```

**Comment by schlessera** [comment]
@mikeyb31 I currently suspect this problem is either specific to your environment or specific to the file you're trying to download (WP 4.8.1 in `en_GB`). 

Did you try using a different locale, just to be able to rule out that the actual en_GB release is broken?

**Comment by gritdigital-mibu** [comment]
Hi @schlessera 

I've just extracted the wordpress-4.8.1-en_GB.zip file manually on the box and the files are all correct and the archive size matches the download available from <https://en-gb.wordpress.org/wordpress-4.8.1-en_GB.zip>

I think you're right about it being environment specific as I did successfully update another project with no problems. I have however seen this problem occur on multiple projects but they both have similar environments.

I just deleted the cached downloads and tried it again but I'm still getting file mismatches and an empty wp-login.php file.

![wp-login-blank](https://user-images.githubusercontent.com/11611416/30582563-0fc9ca46-9d1c-11e7-9600-1303a4a84c34.png)

I'm keen to workout what's going on so I'll feedback to you if I find anything.
Cheers
